src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,164 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
```

Comment:
The use of `Arrays` and `Objects` utility classes is against the checklist guideline: "Don't use Objects, Arrays, or any other util class." Consider implementing the required functionality without these utilities.